lib/cache/cache.go

```diff
@@ -323,6 +324,18 @@ func ForWindowsDesktop(cfg Config) Config {
 	return cfg
 }
 
+// ForDiscovery sets up watch configuration for discovery servers.
+func ForDiscovery(cfg Config) Config {
+	cfg.target = "discovery"
+	cfg.Watches = []types.WatchKind{
+		{Kind: types.KindCertAuthority, LoadSecrets: false},
+		{Kind: types.KindClusterName},
+		{Kind: types.KindNamespace, Name: apidefaults.Namespace},
```

Comment:
Hmm, so shouldn't we include nodes here as well? Otherwise I think we may be going straight to auth server.

I think that we should include nodes here and below, where you initialize node watcher, it should be initialized with the access point (which is cache-based) instead of directly with the auth client.